### PR TITLE
fix wasm_shim allocation behavior as per #248

### PR DESCRIPTION
**Pull request by SFBdragon**

### CODE REVIEW

File: zstd-safe/zstd-sys/src/wasm_shim.rs [hunk 1]
```diff
@@ -14,17 +29,33 @@ pub extern "C" fn rust_zstd_wasm_shim_calloc(
     nmemb: usize,
     size: usize,
 ) -> *mut c_void {
+    // the strategy for rust_zstd_wasm_shim_malloc is used, described there
```
Comment (gyscos) [inline zstd-safe/zstd-sys/src/wasm_shim.rs:32]: Should we maybe just call `rust_zstd_wasm_shim_malloc(nmemb * size)`?

Comment (SFBdragon) [reply zstd-safe/zstd-sys/src/wasm_shim.rs]: Sounds good, although this prompted me to check the c docs: turns out calloc expects its memory to be zeroed.

**Comment by gyscos** [comment]
> Sounds good, although this prompted me to check the c docs: turns out calloc expects its memory to be zeroed.

Woops indeed! Guess on wasm there's no special CoW zero page we could use, so let's just zero things out manually here?

**Comment by SFBdragon** [comment]
I hope you can squash my messy commit history :)

(Everything is still using std for now)

**Pull request merged**